sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1688,6 +1688,12 @@ object SQLConf {
     .stringConf
     .checkValues(Set("Timestamp", "Instant"))
     .createWithDefault("Timestamp")
+
+  val DATE_EXTERNAL_TYPE = buildConf("spark.sql.catalyst.dateType")
```

Comment:
shall we combine the configs? e.g. `spark.sql.datetime.java8API.enabled`

Reply:
I combined them.